src/yoroi-wallets/cardano/ShelleyWallet.ts

```diff
@@ -127,20 +123,24 @@ export class ShelleyWallet implements WalletInterface {
   rewardAddressHex: null | string = null
   version: undefined | string
   checksum: undefined | CardanoTypes.WalletChecksum
+  private _utxos: RawUtxo[]
 
   // =================== create =================== //
-  constructor(storage: typeof storageLegacy, networkId: NetworkId, id: string) {
+  constructor(storage: typeof storageLegacy, networkId: NetworkId, id: string, utxoManager: UtxoManager) {
     this.id = id
     this.storage = storage
     this.networkId = networkId
     this.defaultAsset = getDefaultAssetByNetworkId(this.networkId)
 
-    const config = this.getBackendConfig()
-    this.utxoStorage = makeUtxoStorage(this.storage, `/wallet/${this.id}/utxos`)
-    this.utxoService = initUtxo(this.utxoStorage, `${config.API_ROOT}/`)
+    this.utxoManager = utxoManager
+    this._utxos = utxoManager.initialUtxos
```

Comment:
this makes the utxos available right after opening the wallet (later we can derive the tokens)